datafusion/functions/src/math/mod.rs

```diff
@@ -24,10 +24,14 @@ mod nans;
 make_udf_function!(nans::IsNanFunc, ISNAN, isnan);
 make_udf_function!(abs::AbsFunc, ABS, abs);
 
-make_math_unary_udf!(TanhFunc, TANH, tanh, tanh);
-make_math_unary_udf!(AcosFunc, ACOS, acos, acos);
-make_math_unary_udf!(AsinFunc, ASIN, asin, asin);
-make_math_unary_udf!(TanFunc, TAN, tan, tan);
+make_math_unary_udf!(Log2Func, LOG2, log2, log2, Some(vec![Some(true)]));
```

Comment:
I wonder if we could make a constant or something that would make this easier to read

For example
```rust
/// Single monotonic argument
const MONOTONIC_ONE: Option<Vec<bool>> = Some(vec![Some(true)])
const NON_MONOTONIC: Option<Vec<bool>> = None;
```

And then 

```suggestion
make_math_unary_udf!(Log2Func, LOG2, log2, log2, MONOTONIC_ONE);
```

Perhaps as a follow on PR

Reply:
@alamb 
I tried this readability fix but Rust doesn't allow allocation (which `vec![]` needs) in `const` or `static` items. Global `let` is also not allowed.

The compiler suggested this with static:
```
consider wrapping this expression in `Lazy::new(|| ...)` from the `once_cell` crate: https://crates.io/crates/once_cell
```

We use `once_cell` in the `physical_plan` crate but introducing it to datafusion functions just for this doesn't seem right to me.

Reply:
> We use `once_cell` in the `physical_plan` crate but introducing it to datafusion functions just for this doesn't seem right to me.

That makes sense to me. We would probably have to throw in a `clone()` or something

I thought about this and wrote up another API that would be much nicer https://github.com/apache/arrow-datafusion/issues/9879  (both for this callsite and montonicity in general)